app/controllers/TpsController.scala

```diff
@@ -39,13 +41,32 @@ class TpsController @Inject() (actions:              Actions,
 
   def storeTpsPayments(): Action[TpsPayments] = actions.strideAuthenticateAction.async(parse.json[TpsPayments]) { implicit request =>
 
-    val tpsId: TpsId = TpsId.fresh
+    val updatedPayments: List[TpsPaymentItem] = request.body.payments map (payment => payment.copy(paymentItemId = Some(PaymentItemId.fresh)))
```

Comment:
It's easier if the PaymentId is generated in FE. This is common practice here.

Reply:
I've added a call so that at the start a new id is generated